demos/speech_recognition_demo/python/ctcdecode-numpy/ctcdecode_numpy/yoklm/memory_section.cpp

```diff
@@ -85,15 +94,30 @@ MemorySection MemorySection::get_and_drop_prefix(size_t size) {
   const uint8_t * ptr_orig = ptr_;
   size_ -= size;
   ptr_ += size;
-  return MemorySection(ptr_orig, size, owner_);
+  return MemorySection(ptr_orig, base_ptr_, size, owner_);
 }
 
 void MemorySection::reset() {
   size_ = 0;
-  ptr_ = nullptr;
+  ptr_ = base_ptr_ = nullptr;
   owner_.reset();
 }
 
+void MemorySectionBitArray::set_stride(int stride) {
+  stride_ = stride;
+  // Dividing max(size_t) by 2 to be sure there're no overflow in operator(),
+  // because BitField::offset being an int cannot exceed max(size_t)/2 (when it's non-negative).
```

Comment:
```suggestion
  // because being an int, BitField::offset cannot exceed max(size_t)/2 (when it's non-negative).
```
That's participle phrase in initial position.